### PR TITLE
Shop initial

### PR DESCRIPTION
**Pull request by gosiamaks**
Simple shop app without nice layout yet, but works. I'm not sure how to show in file views/warehouse that the quantity of products reduce while they're added to basket (I made that subpage just to see if it works).

### CODE REVIEW

File: shop.rb [hunk 1]
```diff
@@ -32,23 +33,31 @@ class App < Sinatra::Base
     get "/products/:id" do |id|
       product = FindProduct.new.call(id)
       halt 404 unless product
+      erb :"products/show", locals: { product: product }
     end
 
     get "/warehouse" do
-      products = FetchWarehouseItems.new.call
+      items = FetchWarehouseItems.new.call
+      erb :"warehouse/index", locals: { items: items, title: "Warehouse" }
```
Comment (gosiamaks) [inline shop.rb:41]: I was trying to make here something analogous to "get" which refers to "/basket" but nothing happens; i don't know if it's only the matter of that the action isn't visible, or this service doesn't work. Should I put some another service which refers to warehouse in line 53, in post "/basket" method?

File: lib/model/item.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    def initialize(product_id:, quantity:)
+      @id = set_id
+      @product_id = product_id
+      @quantity = quantity
+    end
+
+    def find_product
+      FindProduct.new.call(product_id)
+      #przywołaj produkty znalezione poprzez service
+      #FindProduct (po ID z tablicy PRODUCTS)
```
Comment (smt116) [inline lib/model/item.rb:17]: Remove this comment or rewrite to english and put just above method definition.

File: lib/model/product.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@ class Product
 
     @@id = 0
 
-    def initialize(name:, price:)
+    def initialize(name, price)
```
Comment (smt116) [inline lib/model/product.rb:7]: How is this model different from `Shop::Item`? You have keyword arguments in its initializer, but here you want to remove them.

File: lib/services/DeleteItemFromBasket.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+require_relative './FetchBasketItems'
+
+module Shop
+  class DeleteItemFromBasket
+    def call(item_id)
+      basket_item = FindBasketItem.new.call(item_id)
+
+      return unless basket_item
+
+      reduce_basket_quantity(basket_item)
```
Comment (smt116) [inline lib/services/DeleteItemFromBasket.rb:10]: `reduce_basket_quantity(basket_item) unless basket_item` and you can remove line 8.

File: lib/services/DeleteItemFromWarehouse.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      warehouse_item = FindWarehouseItem.new.call(item_id)
+
+      return unless warehouse_item
+
+      reduce_warehouse_quantity(warehouse_item)
+    end
+
+    private
+    def reduce_warehouse_quantity(warehouse_item)
+      if warehouse_item.quantity > 1 #jeśli >1 to odejmij 1
```
Comment (smt116) [inline lib/services/DeleteItemFromWarehouse.rb:15]: this comment is useless

File: shop.rb [hunk 6]
```diff
@@ -1,5 +1,63 @@
-require relative './lib/model/product'
+require "bundler/setup"
+require "sinatra/base"
+
+Dir["./lib/**/*.rb"].each{|file| require file}
```
Comment (smt116) [inline shop.rb:4]: missing spacebars around curly brackets.

**Comment by smt116** [comment]
Try to work on your commit messages as they should describe the change. Also some of them could be squashed into a single one.

**Pull request merged**